File: server/src/main/scala/kpn/api/common/monitor/MonitorRouteChangeDetail.scala
package kpn.api.common.monitor

import kpn.api.common.Bounds
import kpn.api.common.changes.details.ChangeKey

case class MonitorRouteChangeDetail(
  key: ChangeKey,
  comment: Option[String],
  wayCount: Long,
  waysAdded: Long,
  waysRemoved: Long,
  waysUpdated: Long,
  osmDistance: Long,
  gpxDistance: Long,
  gpxFilename: String,
  bounds: Bounds,
  referenceJson: String,
  routeSegmentCount: Long,
  routeSegments: Seq[MonitorRouteSegment],
  newNokSegments: Seq[MonitorRouteNokSegment],
  resolvedNokSegments: Seq[MonitorRouteNokSegment],
  happy: Boolean,
  investigate: Boolean
)

File: backend/api/src/main/java/kpn/api/common/monitor/MonitorRouteChangeDetail.java
package kpn.api.common.monitor;

import kpn.api.common.Bounds;
import kpn.api.common.changes.details.ChangeKey;
import kpn.api.common.monitor.MonitorRouteDeviation;
import kpn.api.common.monitor.MonitorRouteSegment;

import java.util.Optional;
import com.google.common.collect.ImmutableList;

public record MonitorRouteChangeDetail(
  ChangeKey key,
  Optional<String> comment,
  Long wayCount,
  Long waysAdded,
  Long waysRemoved,
  Long waysUpdated,
  Long osmDistance,
  Long gpxDistance,
  String gpxFilename,
  Bounds bounds,
  String referenceJson,
  Long routeSegmentCount,
  ImmutableList<MonitorRouteSegment> routeSegments,
  ImmutableList<MonitorRouteDeviation> newDeviations,
  ImmutableList<MonitorRouteDeviation> resolvedDeviations,
  Boolean happy,
  Boolean investigate
) {
}
